File: 3143-longest-unequal-adjacent-groups-subsequence-i/longest-unequal-adjacent-groups-subsequence-i.js
/**
 * @param {string[]} words
 * @param {number[]} groups
 * @return {string[]}
 */
var getLongestSubsequence = function(words, groups) {
    let ans = [];
    let n = groups.length;
    for(let i=0; i<n; i++){
        if(i == 0 || groups[i] != groups[i-1]){
            ans.push(words[i]);
        }
    }
    return ans;
};

File: 3143-longest-unequal-adjacent-groups-subsequence-i/longest-unequal-adjacent-groups-subsequence-i.java
class Solution {
    public List<String> getLongestSubsequence(String[] words, int[] groups) {
        List<String> res = new ArrayList<>();
        int n = groups.length;
        for(int i=0; i<n; i++){
            if(i==0 || groups[i] != groups[i-1]){
                res.add(words[i]);
            }
        }
        return res;
    }
}